sympy/solvers/tests/test_solveset.py

```diff
@@ -2239,6 +2261,9 @@ def test_issue_17906():
     assert solveset(7**(x**2 - 80) - 49**x, x) == FiniteSet(-8, 10)
 
 
+@XFAIL
+# solveset does not handle the case when the solution is a set
+# instead of ImageSet at line 3302 at this time
 def test_issue_17933():
```

Comment:
`solveset`, as indicated, does not know what to do with the FiniteSet that is obtained instead of an ImageSet.